src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -373,92 +359,60 @@ else if (cmdLine.hasOption(OPTION_C_NAME)) {
 
         // create config helper object
-        final CliOptions config = convertCliToPojo(commandLine, filesToProcess);
-        if (commandLine.hasOption(OPTION_T_NAME)) {
+        if (printAst) {
             // print AST
-            final File file = config.files.get(0);
+            final File file = filesToProcess.get(0);
             final String stringAst = AstTreeStringPrinter.printFileAst(file,
                     JavaParser.Options.WITHOUT_COMMENTS);
             System.out.print(stringAst);
         }
-        else if (commandLine.hasOption(OPTION_CAPITAL_T_NAME)) {
-            final File file = config.files.get(0);
+        else if (printAstWithComments) {
+            final File file = filesToProcess.get(0);
             final String stringAst = AstTreeStringPrinter.printFileAst(file,
                     JavaParser.Options.WITH_COMMENTS);
             System.out.print(stringAst);
         }
-        else if (commandLine.hasOption(OPTION_J_NAME)) {
-            final File file = config.files.get(0);
+        else if (printJavadocTree) {
+            final File file = filesToProcess.get(0);
             final String stringAst = DetailNodeTreeStringPrinter.printFileAst(file);
             System.out.print(stringAst);
         }
-        else if (commandLine.hasOption(OPTION_CAPITAL_J_NAME)) {
-            final File file = config.files.get(0);
+        else if (printTreeWithJavadoc) {
+            final File file = filesToProcess.get(0);
             final String stringAst = AstTreeStringPrinter.printJavaAndJavadocTree(file);
             System.out.print(stringAst);
         }
-        else if (commandLine.hasOption(OPTION_S_NAME)) {
-            final File file = config.files.get(0);
-            final String suppressionLineColumnNumber = config.suppressionLineColumnNumber;
-            final int tabWidth = config.tabWidth;
+        else if (hasSuppressionLineColumnNumber) {
+            final File file = filesToProcess.get(0);
             final String stringSuppressions =
                     SuppressionsStringPrinter.printSuppressions(file,
                             suppressionLineColumnNumber, tabWidth);
             System.out.print(stringSuppressions);
         }
         else {
-            if (commandLine.hasOption(OPTION_D_NAME)) {
+            if (debug) {
                 final Logger parentLogger = Logger.getLogger(Main.class.getName()).getParent();
                 final ConsoleHandler handler = new ConsoleHandler();
                 handler.setLevel(Level.FINEST);
-                handler.setFilter(new Filter() {
-                    private final String packageName = Main.class.getPackage().getName();
-
-                    @Override
-                    public boolean isLoggable(LogRecord record) {
-                        return record.getLoggerName().startsWith(packageName);
-                    }
-                });
+                handler.setFilter(new OnlyCheckstyleLoggersFilter());
```

Comment:
What was the matter with the original way the filter was done?

Reply:
One of the checks (I forgot which one) failed with "anonymous inner class can be replaced with static inner class".